src/compose/network.ts

```diff
@@ -0,0 +1,254 @@
+		this.networkOpts = opts;
+	}
+
+	public static fromDockerodeNetwork(
+		opts: NetworkOptions,
+		network: NetworkInspect,
+	): Network {
+
+		const ret = new Network(opts);
+
+		const match = network.Name.match(/^([0-9]+)_(.+)$/);
+		if (match == null) {
+			// FIXME: Type this properly
+			throw new Error('Invalid network name: ' + network.Name);
+		}
+		const appId = checkInt(match[1]) || null;
+		if (!appId) {
+			throw new Error('Invalid appId: ' + appId);
+		}
+
+		ret.appId = appId;
+		ret.name = match[2];
+		ret.config = {
+			driver: network.Driver,
+			ipam: {
+				driver: network.IPAM.Driver,
+				config: _.map(network.IPAM.Config, (conf) => {
+					// Assign this object of the type of NetworkConfig.ipam.config array
+					const newConf: NetworkConfig['ipam']['config'][0] = {
+						subnet: conf.Subnet,
+						gateway: conf.Gateway,
+					};
+
+					// FIXME: Add proper support for this
+					//if (conf.IPRange != null) {
+						//newConf.ipRange = conf.IPRange;
+					//}
+					//if (conf.AuxAddress != null) {
+						//newConf.auxAddress = conf.AuxAddress;
+					//}
+					return newConf;
+				}),
+				options: network.IPAM.Options == null ? { } : network.IPAM.Options,
+			},
+			enableIPv6: network.EnableIPv6,
+			internal: network.Internal,
+			labels: _.omit(network.Labels, [ 'io.resin.supervised' ]),
+			options: network.Options,
+		};
+
+		return ret;
+	}
+
+	public static async fromAppIdAndName(
+		opts: NetworkOptions,
+		name: string,
+		appId: number,
+	): Bluebird<Network> {
+		const network = await opts.docker.getNetwork(`${appId}_${name}`).inspect();
+		return Network.fromDockerodeNetwork(opts, network);
+	}
+
+	public static fromComposeObject(
+		opts: NetworkOptions,
+		name: string,
+		appId: number,
+		// FIXME: Ensure these are always the same!
+		network: NetworkConfig,
+	): Network {
+		const net = new Network(opts);
+		net.name = name;
+		net.appId = appId;
+
+		// Assign the default values for a network inspect,
+		// so when we come to compare, it will match
+		net.config = _.defaultsDeep(network, {
+			driver: 'bridge',
+			ipam: {
+				driver: 'default',
+				config: [],
+				options: { },
+			},
+			enableIPv6: false,
+			internal: false,
+			labels: { },
+			options: { },
+		});
+
+		return net;
+	}
+
+	public create(): Bluebird<void> {
+		this.logger.logSystemEvent(logTypes.createNetwork, { network: { name: this.name } });
+
+		return Network.fromAppIdAndName(this.networkOpts, this.name, this.appId)
+			.then((current) => {
+				if (!this.isEqualConfig(current)) {
+					// FIXME: type this error
+					throw new Error(
+						`Trying to create network '${this.name}', but a network` +
+						' with the same name and different configuration exists',
+					);
+				}
+
+				// We have a network with the same config and name already created -
+				// we can skip this.
+			})
+			.catch(NotFoundError, () => {
+				return this.docker.createNetwork({
+					Name: this.getDockerName(),
+					Driver: this.config.driver,
```

Comment:
I recommend also setting the CheckDuplicate option here (I was planning on doing it in https://github.com/resin-io/resin-supervisor/pull/633/files#diff-d40727b4ebe673427c629249a8bb1db4R108), as it will help avoid some of the weird cases when there's two networks with the same name.